### PR TITLE
chore(go): Add support go 1.20

### PR DESCRIPTION
**Pull request by mstmdev**

### CODE REVIEW

File: .github/workflows/gin.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest, macos-latest]
-        go: [1.16, 1.17, 1.18, 1.19]
+        go: ['1.16', '1.17', '1.18', '1.19', '1.20']
```
Comment (thinkerou) [inline .github/workflows/gin.yml:34]: why add `'`?

Comment (mstmdev) [reply .github/workflows/gin.yml:34]: image

It does not appear to be correctly identified as go1.20 [here](https://github.com/gin-gonic/gin/actions/runs/4071314852/jobs/7013011269).
@thinkerou

**Comment by mstmdev** [comment]
Wait for [sonic](https://github.com/bytedance/sonic) to support go1.20.
https://github.com/bytedance/sonic/issues/355

**Comment by mstmdev** [comment]
It's done, please review.
@thinkerou

**Review by appleboy — APPROVED** [review]

**Comment by appleboy** [comment]
@thinkerou Please help to review.

**Review by thinkerou — APPROVED** [review]

**Pull request merged**